src/credentials/attestation_verifier/DefaultDeviceAttestationVerifier.cpp

```diff
@@ -607,12 +613,147 @@ CHIP_ERROR DefaultDACVerifier::VerifyNodeOperationalCSRInformation(const ByteSpa
+//    "revoked_serial_numbers": [
+//      "69CDF10DE9E54ED1"
+//    ]
+//  }
+// ]
+//
+bool DefaultDACVerifier::IsEntryExistsInRevocationSet(const CharSpan & akidHexStr, const CharSpan & issuerNameBase64Str,
+                                                      const CharSpan & serialNumberHexStr)
+{
+    std::ifstream file(mDeviceAttestationRevocationSetPath);
+    if (!file.is_open())
+    {
+        return false;
+    }
+
+    // Parse the JSON data incrementally
+    Json::CharReaderBuilder readerBuilder;
+    Json::Value jsonData;
+    std::string errs;
+
+    bool parsingSuccessful = Json::parseFromStream(readerBuilder, file, &jsonData, &errs);
+
+    // Close the file as it's no longer needed
+    file.close();
+
+    if (!parsingSuccessful)
+    {
+        return false;
+    }
+
+    for (const auto & revokedSet : jsonData)
+    {
+        if (strncmp(revokedSet["issuer_name"].asCString(), issuerNameBase64Str.data(), issuerNameBase64Str.size()) == 0 &&
+            strncmp(revokedSet["issuer_subject_key_id"].asCString(), akidHexStr.data(), akidHexStr.size()) == 0)
+        {
+            for (const auto & revokedSerialNumber : revokedSet["revoked_serial_numbers"])
+            {
+                if (strncmp(revokedSerialNumber.asCString(), serialNumberHexStr.data(), serialNumberHexStr.size()) == 0)
+                {
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR DefaultDACVerifier::GetAKIDHexStr(const ByteSpan & certDer, MutableCharSpan & outAKIDHexStr)
+{
+    uint8_t akidBuf[kAuthorityKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+
+    CHIP_ERROR err = ExtractAKIDFromX509Cert(certDer, akid);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+    VerifyOrReturnError(outAKIDHexStr.size() > akid.size() * 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    Encoding::HexFlags flags = Encoding::HexFlags::kUppercaseAndNullTerminate;
+    err                      = BytesToHex(akid.data(), akid.size(), outAKIDHexStr.data(), outAKIDHexStr.size(), flags);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+
+    outAKIDHexStr.reduce_size(strlen(outAKIDHexStr.data()));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultDACVerifier::GetSerialNumberHexStr(const ByteSpan & certDer, MutableCharSpan & outSerialNumberHexStr)
+{
+    uint8_t serialNumberBuf[kMaxCertificateSerialNumberLength] = { 0 };
+    MutableByteSpan serialNumber(serialNumberBuf);
+
+    CHIP_ERROR err = ExtractSerialNumberFromX509Cert(certDer, serialNumber);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+    VerifyOrReturnError(outSerialNumberHexStr.size() > serialNumber.size() * 2, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    Encoding::HexFlags flags = Encoding::HexFlags::kUppercaseAndNullTerminate;
+    err = BytesToHex(serialNumber.data(), serialNumber.size(), outSerialNumberHexStr.data(), outSerialNumberHexStr.size(), flags);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, err);
```

Comment:
Extract hex-ification to a helper, with ByteSpan input and returning an optional<std::string>